src/main/java/org/dependencytrack/model/WorkflowState.java

```diff
@@ -0,0 +1,136 @@
+package org.dependencytrack.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import javax.jdo.annotations.Column;
+import javax.jdo.annotations.FetchGroup;
+import javax.jdo.annotations.FetchGroups;
+import javax.jdo.annotations.IdGeneratorStrategy;
+import javax.jdo.annotations.PersistenceCapable;
+import javax.jdo.annotations.Persistent;
+import javax.jdo.annotations.PrimaryKey;
+import javax.validation.constraints.NotNull;
+import java.io.Serializable;
+import java.util.Date;
+import java.util.UUID;
+
+@PersistenceCapable(table= "WORKFLOW_STATE")
+@FetchGroups({
+        @FetchGroup(name = "ALL", members = {
+                @Persistent(name = "id"),
+                @Persistent(name = "parent"),
+                @Persistent(name = "token"),
+                @Persistent(name = "updatedAt"),
+                @Persistent(name = "startedAt"),
+                @Persistent(name = "step"),
+                @Persistent(name = "status")
+        })
+})
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class WorkflowState implements Serializable {
+
+    public enum FetchGroup {
+        ALL
+    }
+
+    @PrimaryKey
+    @Persistent(valueStrategy = IdGeneratorStrategy.NATIVE)
+    private long id;
+
+    // null allowed because first step will have no parent and will be the first step of recursion
+    @Persistent
+    @Column(name = "PARENT_STEP_ID" , allowsNull = "true")
+    private WorkflowState parent;
+
+    @Column(name = "TOKEN", jdbcType = "VARCHAR", length = 36, allowsNull = "false")
+    @NotNull
+    private UUID token;
+
+    @Persistent
+    @Column(name = "STARTED_AT", allowsNull = "true")
+    @NotNull
+    private Date startedAt;
+
+    @Persistent
+    @Column(name = "UPDATED_AT", allowsNull = "true")
+    @NotNull
+    private Date updatedAt;
+
+    @Persistent
+    @Column(name = "STEP", jdbcType = "VARCHAR", length = 64, allowsNull = "false")
+    @NotNull
+    private String step;
+
+    @Persistent
+    @Column(name = "STATUS", jdbcType = "VARCHAR", length = 64, allowsNull = "false")
+    @NotNull
+    private String status;
```

Comment:
I think it'd make sense to use enums for these fields. 

For `status` in particular, this would make it possible to encode the state machine behavior, e.g. which state is terminal, whether it is possible to transition from state A to state B etc.